File: healthy_scanner/lib/models/product.dart
class Product {
  final String id; // UUID
  final String? barcode;
  final String name;
  final String? brand;
  final String? category;

  Product({
    required this.id,
    this.barcode,
    required this.name,
    this.brand,
    this.category,
  });

  // sqflite / 서버 둘 다에서 쓸 수 있게 fromMap/toMap
  factory Product.fromMap(Map<String, dynamic> map) {
    return Product(
      id: map['id'] as String,
      barcode: map['barcode'] as String?,
      name: map['name'] as String,
      brand: map['brand'] as String?,
      category: map['category'] as String?,
    );
  }

  Map<String, dynamic> toMap() {
    return {
      'id': id,
      'barcode': barcode,
      'name': name,
      'brand': brand,
      'category': category,
    };
  }
}

File: app/models/product.py
# app/models/product.py
from sqlalchemy import Column, String, Integer, Text, DateTime
from sqlalchemy.sql import func

from app.core.database import Base
from sqlalchemy.orm import Mapped, mapped_column
from datetime import datetime

class Product(Base):
    __tablename__ = "product"

    id: Mapped[str] = mapped_column(String(36), primary_key=True, index=True)  # CHAR(36)
    barcode: Mapped[str] = mapped_column(String(32), unique=True, nullable=True)
    barcode_kind: Mapped[str] = mapped_column(String(16), nullable=True)       # 'EAN13' | ...

    brand: Mapped[str] = mapped_column(String(128), nullable=True)
    name: Mapped[str] = mapped_column(String(256), nullable=True)
    category: Mapped[str] = mapped_column(String(128), nullable=True)
    size_text: Mapped[str] = mapped_column(String(64), nullable=True)
    image_url: Mapped[str] = mapped_column(Text, nullable=True)
    country: Mapped[str] = mapped_column(String(64), nullable=True)
    notes: Mapped[str] = mapped_column(Text, nullable=True)

    score: Mapped[int] = mapped_column(Integer, nullable=True)

    allergens: Mapped[str] = mapped_column(Text, nullable=True)
    trace_allergens: Mapped[str] = mapped_column(Text, nullable=True)

    created_at: Mapped[datetime] = mapped_column(
        DateTime(timezone=False),
        nullable=False,
        server_default=func.now(),
    )
    updated_at: Mapped[datetime] = mapped_column(
        DateTime(timezone=False),
        nullable=False,
        server_default=func.now(),
        onupdate=func.now(),
    )
